docker/docker.go

```diff
@@ -571,5 +584,71 @@ func Attach(ctx context.Context, config *container.Config, host *container.HostC
 		exit <- code
 	}()
 
+	monitorTtySize(c, res.ID)
+
 	return &resp, exit, nil
 }
+
+func getStdOutSize() (uint, uint) {
+	fd, isTerminal := term.GetFdInfo(os.Stdout)
+	if !isTerminal {
+		return 0, 0
+	}
+
+	ws, err := term.GetWinsize(fd)
+	if err != nil {
+		return 0, 0
+	}
+
+	return uint(ws.Height), uint(ws.Width)
+}
+
+func monitorTtySize(c *client.Client, containerID string) {
+	initTtySize(c, containerID)
+	if runtime.GOOS == "windows" {
+		go func() {
+			prevH, prevW := getStdOutSize()
+			for {
+				time.Sleep(time.Millisecond * 250)
+				h, w := getStdOutSize()
+
+				if prevW != w || prevH != h {
+					resizeTty(c, containerID)
+				}
+				prevH = h
+				prevW = w
+			}
+		}()
+	} else {
+		sigchan := make(chan os.Signal, 1)
+		gosignal.Notify(sigchan, signal.SIGWINCH)
```

Comment:
Is this triggered only for this case?

Reply:
sorry I don't understand your question. This line subscribes to SIGWINCH signal

Reply:
Sorry, I mean is the `SIGWINCH` signal being sent only in case of terminal resize?

Reply:
you can send it manually if you want of course.
But it's how you handle terminal resize on *nix systems https://en.wikipedia.org/wiki/Signal_(IPC)#SIGWINCH

Reply:
Thank you!